puzzle-utils/words/wordsearch.py

```diff
@@ -23,25 +23,32 @@ def wordsearch_reduce(grid, words):
     """
     results = wordsearch_find(grid, words)
 
+    # Create a normalized grid for reference and another for keeping track of used cells
+    reference_grid = []
     working_grid = []
     for row in grid:
-        working_grid.append(list(row))
+        reference_row = [x.lower() for x in list(row)]
+        reference_grid.append(reference_row)
+        working_grid.append(reference_row[:])
 
+    # Remove words from the working grid
     for result in results:
         row = result.row
         column = result.column
         row_increment, column_increment = direction_to_increment_mapping[result.direction]
-        for _ in result.word:
+        while len(result.word) > 0:
+            result.word = result.word.replace(reference_grid[row][column], '', 1)
             working_grid[row][column] = None
             row += row_increment
             column += column_increment
     
+    # Compress the working grid
     reduced_grid = []
     for row in range(len(working_grid)):
         reduced_grid.append('')
-        for char in working_grid[row]:
-            if char != None:
-                reduced_grid[row] += char
+        for cell in working_grid[row]:
+            if cell != None and not cell.isspace():
```

Comment:
It would be nice if you could do this in your normalization pass on the grid to not have to do this extra check